common/src/reserved.rs

```diff
@@ -65,12 +86,129 @@ impl ReservedState {
             .collect())
     }
 
-    pub fn apply_delegate(&mut self, _tx: &TxDelegate) -> Result<Self, String> {
-        unimplemented!()
+    pub fn apply_delegate(&mut self, tx: &TxDelegate) -> Result<Self, String> {
+        let mut delegator_updated = false;
+        let delegator_name = match self.query_name(&tx.delegator.clone()) {
+            Some(name) => name,
+            None => return Result::Err("delegator does not exist by name".to_string()),
+        };
+
+        let delegatee_name = match self.query_name(&tx.delegatee.clone()) {
+            Some(name) => name,
+            None => return Result::Err("delegatee does not exist by name".to_string()),
+        };
+
+        for member in &mut self.members {
+            if member.name == delegator_name {
+                match Self::update_delegator_delegatee_field(tx, &delegatee_name, member) {
```

Comment:
Why make another function? The logic is not that complicated and not that long. Also, it reduces code readability.